src/clusterfuzz/_internal/cron/project_setup.py

```diff
@@ -545,7 +552,15 @@ def create_project_settings(project, info, service_account):
 
 def create_pubsub_topics(project):
   """Create pubsub topics for tasks."""
-  for platform in PUBSUB_PLATFORMS:
+  platforms = PUBSUB_PLATFORMS
+  if 'android_' not in project:
+    platforms = PUBSUB_PLATFORMS
+  else:
+    platforms = [project.replace('android_', '')]
```

Comment:
Yes, per the suggestion above, this was generalized into the two new functions.

The new function creates a queue name directly based on the `queue_id`, however, the downsteam helper function still extracts `project` & `platform` from the `queue_id`. I prefer this implementation since the queue naming logic is heavily dependent on this existing  project / platform combination:

```
def create_pubsub_topics_for_queue_id(queue_id):
  """Create pubsub topics from project configs for tasks."""
  project, platform = queue_id.split('-') # Although we ingest queue_id, still rely on legacy naming convention
  name = untrusted.queue_name(project, platform)
  _create_pubsub_topic(name)
```

Let me know if this looks alright to you